MMOCoreORB/src/server/zone/objects/creature/commands/RequestSetStatMigrationDataCommand.h

```diff
@@ -51,6 +51,31 @@ class RequestSetStatMigrationDataCommand : public QueueCommand {
 			return GENERALERROR;
 		}
 
+		if (creature->isInCombat()) {
+			creature->sendSystemMessage("You cannot migrate stats in combat!");
+			return 0;
```

Comment:
should return `GENERALERROR` here instead of 0 (zero is the same as [`SUCCESS`](https://github.com/TrigsC/Core3/blob/1216d8cde2b7287ff74526f068141a402dec165a/MMOCoreORB/src/server/zone/objects/creature/commands/QueueCommand.h#L57))

```suggestion
			return GENERALERROR;
```

Reply:
Does GENERALERROR return something to player? Reason I went with 0 is for custom error message.

Reply:
Nope. In fact, what you're returning is the error number. Returning 0 means "no error", or SUCCESS. Which is actually not true in this case.

A SUCCESS case causes the [object controller to call onComplete()](https://github.com/Trigsc/Core3/blob/unstable/MMOCoreORB/src/server/zone/managers/objectcontroller/ObjectControllerImplementation.cpp#L159-L166) for the queue command instead of indicating the command failed.

Only a [handful of error codes](https://github.com/TrigsC/Core3/blob/unstable/MMOCoreORB/src/server/zone/objects/creature/commands/QueueCommand.cpp#L103-L207) send default messages. Typically the return values you'll return are:

- `INVALIDTARGET`: whatever they're targeting isn't the right thing. Might not be a player, or isn't a creature, or whatever conditions about their target for the command are not met. No default message is sent, you must send one.
- `INVALIDSYNTAX`: arguments provided are invalid. You can provide an additional message to the creature to clarify, but the syntax help string of the command (if one exists) will be sent to the creature. You can override what the syntax message is in `getSyntax()`
- `GENERALERROR`: anything else that doesn't fit in the above categories. If it makes sense to send a system message, you should.
- `SUCCESS`: things worked normally!

I would always avoid sending a manual error code and always send one of the [enum values provided by QueueCommand.h](https://github.com/TrigsC/Core3/blob/unstable/MMOCoreORB/src/server/zone/objects/creature/commands/QueueCommand.h#L57-L73). 

You can check out those enums as well as QueueCommand.cpp to see what it means when you return them if you're curious about what returning different codes would mean for your command and if it's more appropriate to return one or the other (such as `TOOFAR` for a situation if you're implementing a combat command and checking distance).